crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -3737,13 +3773,100 @@ where
     pub fn spawned_by(&self) -> MaybeLocation {
         self.entity.spawned_by()
     }
+
+    /// Gets the component of the given [`ComponentId`] from the entity.
+    ///
+    /// **You should prefer to use the typed API [`Self::get`] where possible and only
+    /// use this in cases where the actual component types are not known at
+    /// compile time.**
+    ///
+    /// Unlike [`EntityRefExcept::get`], this returns a raw pointer to the component,
+    /// which is only valid while the [`EntityRefExcept`] is alive.
+    #[inline]
+    pub fn get_by_id(&self, component_id: ComponentId) -> Option<Ptr<'w>> {
```

Comment:
I think this could be generalized using `DynamicComponentFetch` to match the other `get_by_id()` methods.  But that should happen anyway when these get merged by the "Pointing Fingers: Deduplicating Entity APIs with EntityPtr" working group, so this is fine for now!